### PR TITLE
add websocket logic to remove polling calls

### PR DESCRIPTION
**Pull request by xavluiz**
* this implements the websocket logic to listen to auth and integration events to remove the polling calls

### CODE REVIEW

File: src/message_handlers/integration_connection.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+	  // update the login status
+	  window.showInformationMessage(`Successfully connected to Spotify. Loading playlists...`);
+
+	  // clear existing spotify integrations
+	  clearSpotifyIntegrations();
+
+	  // update the spotify integrations before populating the spotify user
+	  await updateSpotifyIntegrations(user);
+	  await updateSlackIntegrations(user);
```
Comment (brettmstevens7) [inline src/message_handlers/integration_connection.ts:45]: Is this the Music Time version of the Slack integration being updated here?

File: src/websockets.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+  ws.on("error", function error(e) {
+    console.error("[CodeTime] error connecting to websockets", e);
+  });
+}
+
+function retryConnection() {
+  console.debug("[CodeTime] retrying websockets connecting in 10 seconds");
+
+  retryTimeout = setTimeout(() => {
```
Comment (brettmstevens7) [inline src/websockets.ts:61]: would this benefit from a max retry count?

**Review by brettmstevens7 — APPROVED** [review]

File: src/message_handlers/integration_connection.ts (hunk 1 above)
Comment (xavluiz) [reply src/message_handlers/integration_connection.ts:45]: Good catch! This was there when music time/spoitify was an auth provider and we'd fetch the new slack integrations. I'll remove the slack integrations call here

File: src/websockets.ts (hunk 2 above)
Comment (xavluiz) [reply src/websockets.ts:61]: Yeah, i'm not sure how we should proceed with this. Its the same in code time, but you're right. If we're having issue, it most likely won't ever reconnect. I'll add that. thanks

Comment (xavluiz) [reply src/websockets.ts:68]: updated, thanks again for reviewing that

**Pull request merged**